src/ci/scripts/should-skip-this.sh

```diff
@@ -6,15 +6,19 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+git fetch "https://github.com/$GITHUB_REPOSITORY" "$GITHUB_BASE_REF"
+BASE_COMMIT="$(git merge-base FETCH_HEAD HEAD)"
+
+echo "Searching for toolstate changes between $BASE_COMMIT and $(git rev-parse HEAD)"
 source "$(cd "$(dirname "$0")" && pwd)/../shared.sh"
 
 if [[ -z "${CI_ONLY_WHEN_SUBMODULES_CHANGED+x}" ]]; then
     echo "Executing the job since there is no skip rule in effect"
-elif git diff HEAD^ | grep --quiet "^index .* 160000"; then
+elif git diff "$BASE_COMMIT" | grep --quiet "^index .* 160000"; then
     # Submodules pseudo-files inside git have the 160000 permissions, so when
     # those files are present in the diff a submodule was updated.
     echo "Executing the job since submodules are updated"
-elif git diff --name-only HEAD^ | grep --quiet src/tools/'\(clippy\|rustfmt\)'; then
+elif git diff --name-only "$BASE_COMMIT" | grep --quiet src/tools/'\(clippy\|rustfmt\)'; then
     # There is not an easy blanket search for subtrees. For now, manually list
     # clippy.
```

Comment:
Stack overflow says this should work: https://stackoverflow.com/a/18339297/7669110
```
 elif git diff --name-only "$BASE_COMMIT" | grep --quiet "$(git log | grep git-subtree-dir | cut -d : -f2 | tr -d ' ' | tr '\n' '|' | sed 's/|$//')"; then
```
but that's a lot more complicated, I'll save it for a follow-up.

Reply:
Note that command indicates this list is missing cranelift:
```
> git log | grep git-subtree-dir
    git-subtree-dir: src/tools/rustfmt
    git-subtree-dir: compiler/rustc_codegen_cranelift
    git-subtree-dir: src/tools/clippy
```

Reply:
It's.. not missing cranelift though, really. We don't run cranelift tests in CI today.

I don't think we should be doing anything fancy to detect subtrees; manual enumeration is perfectly acceptable - we don't regularly add or remove them. The BASE_COMMIT there seems like a good start though.